spec/models/miq_worker/container_common_spec.rb

```diff
@@ -113,6 +113,35 @@ def deployment_name_for(name)
         expect(klass.constantize.new.worker_deployment_name.length).to be <= 60
       end
     end
+
+    context "ems_id" do
+      let(:subject) { ManageIQ::Providers::Openshift::ContainerManager::EventCatcher.new }
+
+      it "is appended" do
+        subject.queue_name = "ems_1"
+        expect(subject.worker_deployment_name[-1]).to eq("1")
+      end
+
+      it "isn't appended for nil queue_name" do
+        subject.queue_name = nil
+        expect(subject.worker_deployment_name[-7, 7]).to eq("catcher")
+      end
+
+      it "isn't appended for blank queue_name" do
+        subject.queue_name = " "
+        expect(subject.worker_deployment_name[-7, 7]).to eq("catcher")
+      end
+
+      it "isn't appended for invalid queue_name prefix" do
+        subject.queue_name = "notems_1"
+        expect(subject.worker_deployment_name[-7, 7]).to eq("catcher")
+      end
+
+      it "isn't appended for array queue_name" do
+        subject.queue_name = %w[ems_1 ems2]
```

Comment:
This is the only case I'm not understanding.  If we have an "Array" queue_name, then the worker doesn't have an id?  what happens if you have 2 separate queue_names like `%w[ems_1 ems2]` and `%w[ems_3 ems4]`?  Wouldn't those collide?

Reply:
I questioned that too.  An array queue_name isn't supported as it was dropped in https://github.com/ManageIQ/manageiq/pull/20345.  

I could just drop this test as it was just confirmation of how to fix the current problem.  This deployment name creation code is so far from the source problem that raising in the code here would be quite unhelpful.

Reply:
Ok, so that's telling me that foreman has a bug and needs to be fixed